src/mobility/use-benefits.tsx

```diff
@@ -0,0 +1,49 @@
+import {useEffect, useState} from 'react';
+import {getBenefits, getValueCode} from './api/api';
+import {useOperators} from '@atb/mobility/use-operators';
+import {
+  OperatorBenefitIdType,
+  OperatorBenefitType,
+} from '@atb-as/config-specs/lib/mobility-operators';
+import {Linking} from 'react-native';
+
+export const useBenefits = (operatorId: string | undefined) => {
+  const operators = useOperators();
+  const [userBenefits, setUserBenefits] = useState<OperatorBenefitIdType[]>([]);
+  const [valueCode, setValueCode] = useState<string | undefined>();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    if (operatorId) {
+      setLoading(true);
+      Promise.all([
+        getBenefits()
+          .then((benefits) => benefits.find((b) => b.operator === operatorId))
+          .then((benefits) => setUserBenefits(benefits?.benefits ?? [])),
```

Comment:
I'm a little confused by the naming here. We got an `benefits` array, and after finding one item in the array it is still called `benefits`, and then `benefits.benefits` is called. So that makes it three layers of `benefits`? 😅 Is it possible to differentiate about what they are, or what level they are on?